medcat/cat.py

```diff
@@ -467,64 +466,64 @@ def _print_stats(self,
 
     def train(self,
               data_iterator: Iterable,
+              nepochs: int = 1,
               fine_tune: bool = True,
               progress_print: int = 1000,
               checkpoint: Optional[Checkpoint] = None,
-              resume_from_checkpoint: bool = False) -> None:
+              is_resumed: bool = False) -> None:
         """ Runs training on the data, note that the maximum length of a line
         or document is 1M characters. Anything longer will be trimmed.
 
         Args:
             data_iterator (Iterable):
                 Simple iterator over sentences/documents, e.g. a open file
                 or an array or anything that we can use in a for loop.
+            nepochs (int):
+                Number of epochs for which to run the training.
             fine_tune (bool):
                 If False old training will be removed.
             progress_print (int):
                 Print progress after N lines.
             checkpoint (Optional[medcat.utils.checkpoint.Checkpoint]):
```

Comment:
I think the save_dir and save_steps should be in the config (or as part of training arguments) and not part of the checkpoint object, it is not intuitive that to create (or even disable) checkpoints I need to provide a checkpoint object.

Reply:
I can see https://github.com/CogStack/MedCAT/pull/202 has started doing that. Shall we make the suggested change on that one?

Reply:
I'll remove 202 and we just make the change here